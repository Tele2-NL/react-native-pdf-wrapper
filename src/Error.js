import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';

const Error = ({ retryPdf }) => (
  <View style={styles.container}>
    <Text style={styles.error}>
      There was an error while trying to load pdf.
    </Text>

    <TouchableOpacity
      onPress={retryPdf}
      style={styles.retryButtonContainer}
    >
      <Text style={styles.retry}>Retry</Text>
    </TouchableOpacity>
  </View>
);

Error.propTypes = {
  retryPdf: PropTypes.func,
};

Error.defaultProps = {
  retryPdf: () => null,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 48,
  },
  error: {
    textAlign: 'center',
    color: 'black',
    fontSize: 14,
  },
  retryButtonContainer: {
    marginTop: 12,
    padding: 6,
    borderRadius: 4,
  },
  retry: {
    textAlign: 'center',
    fontWeight: 'bold',
  },
});

export default Error;
