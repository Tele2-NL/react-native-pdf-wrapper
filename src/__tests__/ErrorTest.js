import React from 'react';
import { shallow } from 'enzyme';
import Error from '../Error';

describe('<Error />', () => {
  beforeEach(() => {
    this.props = {
      retryPdf: jest.fn(),
    };

    this.wrapper = shallow(<Error {...this.props} />);
  });

  it('renders correctly', () => {
    expect(this.wrapper).toMatchSnapshot();
  });

  it('should call `retryPdf` when pressing the button', () => {
    this.wrapper.find('TouchableOpacity').first().props().onPress();

    expect(this.props.retryPdf).toHaveBeenCalledTimes(1);
  });
});
