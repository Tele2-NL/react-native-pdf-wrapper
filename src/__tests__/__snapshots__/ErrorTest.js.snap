// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 48,
    }
  }
>
  <Component
    style={
      Object {
        "color": "black",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    There was an error while trying to load pdf.
  </Component>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[MockFunction]}
    style={
      Object {
        "borderRadius": 4,
        "marginTop": 12,
        "padding": 6,
      }
    }
  >
    <Component
      style={
        Object {
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Retry
    </Component>
  </TouchableOpacity>
</Component>
`;
